File: minierp.client/src/mappers/customerMapper.js
export const mapCustomerDtoToForm = (customerDto) => {
    return {
        id: customerDto.id || "",
        name: customerDto.name || "",
        email: customerDto.email || "",
        street: customerDto.street || "",
        city: customerDto.city || "",
        zip: customerDto.zip || "",
        phone: customerDto.phone || "",
        ico: customerDto.ico || "",           
        dic: customerDto.dic || "",
        accountNumber: customerDto.accountNumber || "",
        bankCode: customerDto.bankCode || "",
        isSupplier: customerDto.isSupplier ?? false,
        note: customerDto.note || "",
    };
};

export const mapFormToCreateCustomerDto = (formData) => {
    return {
        name: formData.name,
        email: formData.email,
        street: formData.street,
        city: formData.city,
        zip: formData.zip,
        phone: formData.phone,
        ico: formData.ico,
        dic: formData.dic,
        accountNumber: formData.accountNumber,
        bankCode: formData.bankCode,
        isSupplier: formData.isSupplier,
        note: formData.note,
    };
};

export const mapFormToUpdateCustomerDto = (formData) => {
    const dto = {};

    if (formData.name) dto.name = formData.name;
    if (formData.email) dto.email = formData.email;
    if (formData.street) dto.street = formData.street;
    if (formData.city) dto.city = formData.city;
    if (formData.zip) dto.zip = formData.zip;
    if (formData.phone) dto.phone = formData.phone;
    if (formData.ico) dto.ico = formData.ico;
    if (formData.dic) dto.dic = formData.dic;
    if (formData.accountNumber) dto.accountNumber = formData.accountNumber;
    if (formData.bankCode) dto.bankCode = formData.bankCode;
    if (formData.isSupplier !== undefined) dto.isSupplier = formData.isSupplier;
    if (formData.note !== undefined) dto.note = formData.note;

    return dto;
};

File: MiniERP.Server/Mappers/CustomerMapper.cs
using MiniERP.Server.DTOs;
using MiniERP.Server.Models;

namespace MiniERP.Server.Mappers;

public static class CustomerMapper
{
    public static CustomerDTO ToDto(Customer c) => new()
    {
        Id = c.Id,
        Name = c.Name,
        Street = c.Street,
        City = c.City,
        Zip = c.Zip,
        Email = c.Email,
        Phone = c.Phone,
        Ico = c.Ico,
        Dic = c.Dic,
        AccountNumber = c.AccountNumber,
        BankCode = c.BankCode,
        IsSupplier = c.IsSupplier,
        Note = c.Note
    };

    public static CustomerDetailDTO ToDetailDto(Customer c) => new()
    {
        Id = c.Id,
        Name = c.Name,
        Street = c.Street,
        City = c.City,
        Zip = c.Zip,
        Email = c.Email,
        Phone = c.Phone,
        Ico = c.Ico,
        Dic = c.Dic,
        AccountNumber = c.AccountNumber,
        BankCode = c.BankCode,
        IsSupplier = c.IsSupplier,
        Note = c.Note,
        Orders = c.Orders.Select(OrderMapper.ToDto).ToList()
    };

    public static Customer ToEntity(CreateCustomerDTO dto) => new()
    {
        Name = dto.Name,
        Street = dto.Street,
        City = dto.City,
        Zip = dto.Zip,
        Email = dto.Email,
        Phone = dto.Phone,
        Ico = dto.Ico,
        Dic = dto.Dic,
        AccountNumber = dto.AccountNumber,
        BankCode = dto.BankCode,
        IsSupplier = dto.IsSupplier,
        Note = dto.Note
    };

    public static void ApplyUpdate(Customer customer, UpdateCustomerDTO dto)
    {
        if (!string.IsNullOrEmpty(dto.Name)) customer.Name = dto.Name;
        if (!string.IsNullOrEmpty(dto.Street)) customer.Street = dto.Street;
        if (!string.IsNullOrEmpty(dto.City)) customer.City = dto.City;
        if (!string.IsNullOrEmpty(dto.Zip)) customer.Zip = dto.Zip;
        if (!string.IsNullOrEmpty(dto.Email)) customer.Email = dto.Email;
        if (!string.IsNullOrEmpty(dto.Phone)) customer.Phone = dto.Phone;
        if (!string.IsNullOrEmpty(dto.Ico)) customer.Ico = dto.Ico;
        if (!string.IsNullOrEmpty(dto.Dic)) customer.Dic = dto.Dic;
        if (!string.IsNullOrEmpty(dto.AccountNumber)) customer.AccountNumber = dto.AccountNumber;
        if (!string.IsNullOrEmpty(dto.BankCode)) customer.BankCode = dto.BankCode;
        if (dto.IsSupplier.HasValue) customer.IsSupplier = dto.IsSupplier.Value;
        if (dto.Note != null) customer.Note = dto.Note;
    }
}
